src/TestFramework/AbstractTestFrameworkAdapter.php

```diff
@@ -190,6 +176,34 @@ private function getCommandLine(
     ): array {
         $frameworkArgs = $this->argumentsAndOptionsBuilder->build($configPath, $extraOptions);
 
-        return $this->commandLineBuilder->build($this->testFrameworkExecutable, $phpExtraArgs, $frameworkArgs);
+        return $this->commandLineBuilder->build(
+            $this->testFrameworkExecutable,
+            $phpExtraArgs,
+            $frameworkArgs
+        );
+    }
+
+    /**
+     * @throws UnsupportedTestFrameworkVersion
+     */
+    private function retrieveVersion(): string
+    {
+        $testFrameworkVersionExecutable = $this->commandLineBuilder->build(
+            $this->testFrameworkExecutable,
+            [],
+            ['--version']
+        );
+
+        $process = new Process($testFrameworkVersionExecutable);
+        $process->mustRun();
+
+        try {
+            return $this->versionParser->parse($process->getOutput());
+        } catch (InvalidVersion $exception) {
+            throw new UnsupportedTestFrameworkVersion(sprintf(
```

Comment:
change here from the refactoring: `VersionParser` now throws a friendly runtime-error instead of an `InvalidArgumentException` (which should never be caught - it's a logic exception). We now catch it to throw a more friendly error